assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -64,17 +68,28 @@ const usePersistedStateReducer = (): [State, Dispatch] => {
   }, [selectedRouteIds, ladderDirections, ladderCrowdingToggles])
 
   useEffect(() => {
-    if (routeTabsToPush && !routeTabsPushInProgress) {
+    if (routeTabsToPush[0] && !routeTabsPushInProgress) {
       dispatch(startingRouteTabsPush())
-      putRouteTabs(routeTabsToPush)
+      putRouteTabs(routeTabsToPush[0])
         .then((response) => {
           if (response.ok) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
+            dispatch(routeTabsPushComplete())
```

Comment:
Good question... I'd like to have it separate just because of how unwieldy the state reducer in general is getting, and I don't _think_ there are any race conditions... but of course I could be wrong. Will think about it.

Reply:
I guess the case where an issue would come up would be if the `useEffect` function got partway through executing but was then interrupted (perhaps due to the dependencies changing), but I don't think React would behave that way? In any event I've never heard anything about having to assume that the function you give to `useEffect` might not run in an atomic fashion.